qutebrowser/app.py

```diff
@@ -343,15 +343,12 @@ def on_focus_changed(_old, new):
         log.misc.debug("on_focus_changed called with non-QWidget {!r}".format(
             new))
 
-    if new is None or not isinstance(new, mainwindow.MainWindow):
-        try:
-            objreg.delete('last-focused-main-window')
-        except KeyError:
-            pass
-        qApp.restoreOverrideCursor()
-    else:
-        objreg.register('last-focused-main-window', new.window(), update=True)
+    if new is not None:
```

Comment:
You removed the `isinstance` check here, which means the mouse cursor would remain hidden when switching focus to a dialog box or something similar.

I'd propose the following:

```python
if not isinstance(new, mainwindow.MainWindow):   # includes an implicit None check anyways
    qApp.restoreOverrideCursor()
window = qApp.activeWindow()
if window is not None:
    objreg.register(...)
```